### PR TITLE
limit report line number

### PR DESCRIPTION
**Pull request by Bertk**
report line number shall not be greater than the existing max. line number
(see also #600).

### CODE REVIEW

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxUtils.java [hunk 1]
```diff
@@ -78,8 +82,32 @@ public static String normalizePathFull(String filename, String baseDir) {
...
+   * @return returns true for Reactor Projects
+   */
   public static boolean isReactorProject(Project project) {
     return project.isRoot() && !project.getModules().isEmpty();
   }
+  
+  /**
+   * @return returns number of Lines of a File
+   */
+  public static int countLines(String normalPath) {
```
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxUtils.java]: Is there no core metric we can use to get the line number?

Comment (Bertk) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxUtils.java]: There is a line number in the size section of the metrics.
http://docs.sonarqube.org/display/SONAR/Metric+definitions#Metricdefinitions-Size
I am not sure whether the line metric is available when the report is consumed.

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxReportSensor.java [hunk 2]
```diff
@@ -239,6 +240,21 @@ private boolean saveViolation(Project project, SensorContext context, String rul
           = org.sonar.api.resources.File.fromIOFile(new File(normalPath), project);
```
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxReportSensor.java]: fromIOFile is deprecated, see http://javadocs.sonarsource.org/4.5.2/apidocs/deprecated-list.html. So we should use something like:
```Java
InputFile file= fs.inputFile(fs.predicates().is(new File(filename)));
int maxLine = file.lines();
```
InputFile has method lines.

Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxReportSensor.java]: performance needs to be measured...

**Pull request closed without merging**